src/System.Private.CoreLib/shared/System/MemoryExtensions.Trim.cs

```diff
@@ -0,0 +1,781 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        public static Memory<T> Trim<T>(this Memory<T> memory, T trimElement)
+            where T : IEquatable<T>
+        {
+            Span<T> span = memory.Span;
```

Comment:
Nit: I think it'd be a bit easier on the JIT if this were ReadOnlySpan instead of Span, such that there's only one cast (here) instead of two (when calling ClampStart and ClampEnd).